core/src/context/backend.rs

```diff
@@ -360,29 +261,31 @@ impl<'a> BackendContext<'a> {
         // Not ideal
         let tree = self.0.tree.clone();
 
-        apply_inline_inheritance(self.0, &tree);
+        inline_inheritance_system(self.0, &tree);
 
         if self.0.style.needs_restyle {
-            apply_styles(self.0, &tree);
+            style_system(self.0, &tree);
```

Comment:
Will the borrow checker let us move the check and the needs_restyle thing into the system call? it would be nice to make this just a list of calls.